crates/egui/src/containers/popup.rs

```diff
@@ -253,18 +253,32 @@ pub fn was_tooltip_open_last_frame(ctx: &Context, widget_id: Id) -> bool {
     })
 }
 
+/// Determines popup's close behavior
+#[derive(Clone, Copy)]
+pub enum PopupCloseBehavior {
```

Comment:
For completeness, I suggest we also add a third alternative called e.g. `IgnoreClicks` which ignores any clicks. You can only close it by pressing Escape or calling `Memory::close_popup` (and maybe later with `Ui::close_menu`).

----

Alternatively we replace this `enum PopupCloseBehavior` with 

```rs
struct PopupOptions {
    pub close_on_click_inside: bool,
    pub close_on_click_outside: bool,
    pub close_on_escape: bool,
}

impl PopupOptions {
    pub const CLOSE_ON_CLICK_ANYWHERE: Self = …;
    pub const CLOSE_ON_CLICK_INSIDE: Self = …;
}
```

That makes it much easier to expand with more options in the future